### PR TITLE
Feature/13 port scraper to python

### PR DESCRIPTION
**Pull request by nugycal**

### CODE REVIEW

**Comment by maxwowo** [comment]
Can you make the scraper use virtual environment

File: scraper/class_scraper.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from bs4 import BeautifulSoup
+import requests
+import re
+
+BASEURL = "http://timetable.unsw.edu.au/2020/COMP1511.html"
```
Comment (maxwowo) [inline scraper/class_scraper.py:5]: Can you name this `BASE_URL` instead

File: scraper/class_scraper.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+def getClassInfo(soup):
```
Comment (maxwowo) [inline scraper/class_scraper.py:7]: Can you use snake case 🐍

**Comment by nugycal** [comment]
> Can you use snake case instead of camel case to be stylish

How is that stylish?

**Comment by maxwowo** [comment]
> 
> 
> > Can you use snake case instead of camel case to be stylish
> 
> How is that stylish?

image
https://www.python.org/dev/peps/pep-0008/
Good practice according to the PEP 8 style guide

File: scraper/new_scraper/class_scraper.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+from bs4 import BeautifulSoup
+import requests
+import re
+
+BASE_URL = "http://timetable.unsw.edu.au/2020/COMP1511.html"
+
```
Comment (maxwowo) [inline scraper/new_scraper/class_scraper.py:6]: 2 blank lines

File: scraper/new_scraper/scraper.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+from bs4 import BeautifulSoup
+import re
+import requests
+
+BASE_URL = "http://timetable.unsw.edu.au/2020/"
+
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:8]: 2 blank lines

File: scraper/new_scraper/scraper.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    for heading in minor_headings:
+        campus = heading.get_text()
+        if campus != '':
+            campuses.append(heading)
+
+    # Find subject area for each campus
+    for campus in campuses:
+        # Find the table containing subject data relative to the campus heading
+        parent_table = campus.parent.find_next('tr')
+        heading = parent_table.find('td', {'class' : 'tableHeading'})
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:35]: Delete white space between 'class' and :

File: scraper/new_scraper/scraper.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+        # Get all table entries containing subject data and store into `entries`
+        low = heading.parent.parent.find_all('tr', {'class' : 'rowLowlight'})
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:38]: Delete white space between 'class' and :

File: scraper/new_scraper/scraper.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        high = heading.parent.parent.find_all('tr', {'class' : 'rowHighlight'})
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:39]: Delete white space between 'class' and :

File: scraper/new_scraper/scraper.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        entries = low + high
+
+        # Construct refined data and store in list relative to campus name in `refinedData`
+        campus_title = campus.getText().strip()
+        refined_data[campus_title] = []
+
+        for entry in entries:
+            row_data = {}
+            cells = entry.find_all('td', {'class' : 'data'})
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:48]: Delete white space between 'class' and :

File: scraper/new_scraper/scraper.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    return refined_data
+
+
+data = requests.get(BASE_URL)
+data = data.text
+
+soup = BeautifulSoup(data, 'html.parser')
+
+print(get_campus_and_subject_areas(soup))
```
Comment (maxwowo) [inline scraper/new_scraper/scraper.py:72]: Wrap inside main function?

File: scraper/new_scraper/class_scraper.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        if re.search("Detail", section):
+            sessions.append(heading)
+
+    for session in sessions:
+        parent_table = session.parent.find_next('table')
+        class_forms = parent_table.find_all('td', {'class': 'formBody'})
+        for class_form in class_forms:
+            rows = class_form.find_all('tr')
+            for row in rows:
+                URL_data = row.find_all('td', {'class': ['label', 'data']})
```
Comment (maxwowo) [inline scraper/new_scraper/class_scraper.py:24]: Let's rename this variable to `url_data` to comply with the style guide. 
image

**Pull request closed without merging**